File: android/src/main/java/com/yugioh/android/base/InnerFragment.java
package com.yugioh.android.base;

import android.app.Fragment;
import android.os.Bundle;
import android.view.*;
import android.view.ViewTreeObserver.OnGlobalLayoutListener;

public abstract class InnerFragment extends Fragment implements OnGlobalLayoutListener, InnerIntf {

    protected View innerView = null;
    protected Bundle innerBundle = null;
    protected String tabTitle;
    protected int tabIcon = -1;

    public InnerFragment() {
        super();
    }

    public InnerFragment(String tabTitle) {
        super();
        this.tabTitle = tabTitle;
    }

    public String getTabTitle() {
        return tabTitle;
    }

    public int getIcon() {
        return tabIcon;
    }

    public void setIcon(int res) {
        this.tabIcon = res;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        innerView = inflater.inflate(getFragmentLayoutResId(), container, false);
        initComponents();
        initEvents();
        innerView.getViewTreeObserver().addOnGlobalLayoutListener(this);
        return innerView;
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        innerBundle = getArguments();
        initLogic();
        if (getActivity().getActionBar() != null) {
            if (getCustomTitle() == null || getCustomTitle().equals("")) {
                if (getBarTitle() != 0) {
                    getActivity().getActionBar().setTitle(getBarTitle());
                }
            } else {
                getActivity().getActionBar().setTitle(getCustomTitle());
            }
        }
    }

    public void setNewArguments(Bundle bn) {
        innerBundle = getArguments();
        onGetNewArguments(bn);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(true);
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        if (getActivity() == null) {
            return;
        }
        if (getActivity().getClass().getName().equals(getMainActivityName())) {
            return;
        }

        initMenu(menu);
    }

    @Override
    public void onGlobalLayout() {
        onLayoutReady();
    }

    protected void onLayoutReady() {

    }

}

File: android/src/main/kotlin/com/yugioh/android/base/InnerFragment.kt
package com.yugioh.android.base

import android.app.Fragment
import android.os.Bundle
import android.view.*
import android.view.ViewTreeObserver.OnGlobalLayoutListener

abstract class InnerFragment: Fragment, ViewTreeObserver.OnGlobalLayoutListener, IIntf {

    protected var innerView: View? = null
    protected var innerBundle: Bundle? = null
    var tabTitle: String? = null
    var tabIcon = -1

    constructor(): super()
    constructor(tabTitle: String): super() {
        this.tabTitle = tabTitle
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (getFragmentLayoutResId() != 0) {
            innerView = inflater?.inflate(getFragmentLayoutResId(), container, false)
        } else {
            innerView = getFramgmentLayoutView()
        }
        initComponents()
        initEvents()
        innerView?.viewTreeObserver?.addOnGlobalLayoutListener(this)
        return innerView
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        innerBundle = innerBundle ?: arguments
        initLogic()
        if (activity.actionBar != null) {
            if (getCustomTitle() == null || getCustomTitle() == "") {
                if (UIInstance.dualPane) {
                    if (getBarTitleWithPath() != 0) {
                        activity.actionBar.setTitle(getBarTitleWithPath())
                    }
                } else {
                    if (getBarTitle() != 0) {
                        activity.actionBar.setTitle(getBarTitle())
                    }
                }
            } else {
                activity.actionBar.title = getCustomTitle()
            }
        }
    }

    open fun setNewArguments(bn: Bundle?) {
        innerBundle = bn
        onGetNewArguments(bn)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }

    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
        if (activity == null) {
            return
        }
        if (activity.javaClass.name == getMainActivityName() && !UIInstance.dualPane) {
            return
        }
        initMenu(menu)
    }

    /**
     * do NOT override this method, use @onLayoutReady instead
     */
    override fun onGlobalLayout() = onLayoutReady()

    /**
     * override the method if you want to re-layout after system layouted
     */
    protected open fun onLayoutReady() { }

    /**
     * override the method if you do not need a layout from resource and @getFragmentLayoutResId returns 0
     */
    protected open fun getFramgmentLayoutView(): View? = null
}